exploration-module/src/main/java/io/edpn/backend/exploration/infrastructure/kafka/sender/RequestDataMessageKafkaSenderService.java

```diff
@@ -0,0 +1,76 @@
+package io.edpn.backend.exploration.infrastructure.kafka.sender;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.edpn.backend.exploration.domain.model.RequestDataMessage;
+import io.edpn.backend.exploration.domain.repository.RequestDataMessageRepository;
+import io.edpn.backend.exploration.domain.service.KafkaSenderService;
+import io.edpn.backend.exploration.infrastructure.kafka.KafkaTopicHandler;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.kafka.core.KafkaTemplate;
+
+import java.util.Objects;
+import java.util.concurrent.Semaphore;
+
+@RequiredArgsConstructor
+@Slf4j
+public class RequestDataMessageKafkaSenderService implements KafkaSenderService<RequestDataMessage> {
+
+    private final ObjectMapper objectMapper;
+
+    private final RequestDataMessageRepository requestDataMessageRepository;
+
+    private final KafkaTopicHandler kafkaTopicHandler;
+    private final KafkaTemplate<String, JsonNode> jsonNodekafkaTemplate;
+
+    private final Semaphore semaphore = new Semaphore(1);
+
+    @Override
+    public void sendToKafka(RequestDataMessage requestDataMessage) {
+        try {
+            semaphore.acquire();
+            if (requestDataMessageRepository.find(requestDataMessage).isEmpty()) {
+                requestDataMessageRepository.create(requestDataMessage);
+            } else {
+                log.debug("info request already queued");
+            }
+
+            sendPendingMessages();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            log.error("Failed to acquire semaphore", e);
+        } finally {
+            semaphore.release();
```

Comment:
Probably there's a reason for this, but from an uninformed POV looks like the `Semaphore` is unnecessarily complicating things and synchronizing on the method would have the same effect?